src/views/v3/Bridge/AssetPicker/ChainList.tsx

```diff
@@ -0,0 +1,231 @@
+        borderRadius: '8px',
+      },
+    }),
+    [theme],
+  );
+
+  const {
+    chainList,
+    selectedChainConfig,
+    showSearch,
+    setShowSearch,
+    onChainSelect,
+  } = props;
+
+  const [topChains, showMoreButton] = useMemo(() => {
+    const allChains = chainList ?? [];
+    const selectedChain = selectedChainConfig;
+
+    // Find the selected chain in supported chains
+    const selectedChainIndex = allChains.findIndex((chain) => {
+      return chain.sdkName === selectedChain?.sdkName;
+    });
+    // If the selected chain is outside the top list, we add it to the top;
+    // otherwise we do not change its index in the top list
+    if (
+      selectedChain &&
+      selectedChainIndex &&
+      selectedChainIndex >= SHORT_LIST_SIZE
+    ) {
+      return [
+        [selectedChain, ...allChains.slice(0, SHORT_LIST_SIZE - 1)],
+        allChains.length > SHORT_LIST_SIZE,
+      ];
+    }
+
+    return [
+      allChains.slice(0, SHORT_LIST_SIZE),
+      allChains.length > SHORT_LIST_SIZE,
+    ];
+  }, [chainList, selectedChainConfig]);
+
+  const shortList = useMemo(() => {
+    return (
+      <List component={Stack} direction="row" data-testid="chain-short-list">
+        {topChains.map((chain: ChainConfig) => (
+          <Tooltip key={chain.sdkName} title={chain.displayName}>
+            <ListItemButton
+              selected={selectedChainConfig?.sdkName === chain.sdkName}
+              sx={styles.chainButton}
+              data-testid={`chain-button-${chain.sdkName.toLowerCase()}`}
+              onClick={() => onChainSelect(chain.sdkName)}
+            >
+              <ChainIcon icon={chain.icon} />
+              <Typography
+                fontSize="12px"
+                lineHeight="12px"
+                marginTop="8px"
+                whiteSpace="nowrap"
+              >
+                {chain.symbol}
+              </Typography>
+            </ListItemButton>
+          </Tooltip>
+        ))}
+
+        {showMoreButton ? (
+          <ListItemButton
+            sx={styles.chainButton}
+            onClick={() => {
+              setShowSearch(true);
+            }}
+          >
+            <PlusIcon sx={{ height: '36px', width: '36px' }} />
+            <Typography
+              fontSize="12px"
+              lineHeight="12px"
+              marginTop="8px"
+              whiteSpace="nowrap"
+            >
+              other
+            </Typography>
+          </ListItemButton>
+        ) : null}
+      </List>
+    );
+  }, [
+    styles.chainButton,
+    onChainSelect,
+    selectedChainConfig?.sdkName,
+    setShowSearch,
+    showMoreButton,
+    topChains,
+  ]);
+
+  const searchList = useMemo(
+    () => (
+      <SearchableList<ChainConfig>
+        searchPlaceholder="Search for a chain"
+        sx={styles.chainSearch}
+        items={chainList ?? []}
+        searchQuery={chainSearchQuery}
+        onQueryChange={setChainSearchQuery}
+        filterFn={(chain, query) =>
```

Comment:
Here and other places, should externalize and memoize (useCallback) as needed, easier to put breakpoints in non-JSX code + JSX remains concise